### PR TITLE
Add typed Edge Registry APIs for listing schemas, thing descriptions, and thing models in the Rust and .NET SDKs.

### PR DESCRIPTION
**Pull request by bilalsellak**
Add typed Edge Registry APIs for listing schemas, thing descriptions, and thing models in the Rust and .NET SDKs.

## Changes

- Add `list_schemas` / `ListSchemasAsync`.
- Add `list_thing_descriptions` / `ListThingDescriptionsAsync`.
- Add `list_thing_models` / `ListThingModelsAsync`.
- Support default, all, and specific group selection.
- Support resource label filtering.
- Add .NET integration coverage for the new APIs.
- Add a Rust Edge Registry example with retry handling for eventual consistency.

### Local integration

- Ran the Rust SDK sample against a local MQTT broker and Edge Registry test host.
- Ran three .NET Edge Registry integration tests against the same host.
- Built and tested TinyKube and Azure-NBC using local SDK path dependencies.

### Standalone WASM end-to-end test

### CODE REVIEW

File: rust/azure_iot_operations_services/src/edge_registry/client.rs [hunk 1]
```diff
@@ -983,6 +983,37 @@ impl Client {
         Ok(response.payload.into())
     }
 
+    /// List the XIDs of the xRegistry Schemas matching the provided constraints.
```
Comment (vaavva) [inline rust/azure_iot_operations_services/src/edge_registry/client.rs]: Tiniest nit for consistency - can you put these at the top of their extension sections, since they are the operation for the highest layer (group/resource/version)? That way they stay in the same order as the Core APIs

File: rust/azure_iot_operations_services/examples/edge_registry_client.rs [hunk 2]
```diff
@@ -0,0 +1,362 @@
...
+    if exit_handle.try_exit().is_err() {
+        exit_handle.force_exit();
+    }
+
+    operation_result?;
+    shutdown_result?;
+    Ok(())
+}
+
+async fn graph_round_trip_if_configured(
```
Comment (vaavva) [inline rust/azure_iot_operations_services/examples/edge_registry_client.rs]: I'm not sure this should be in an SDK sample - seems wasm specific? We want these to be as simple as possible to just show the SDK capabilities

File: rust/azure_iot_operations_services/examples/edge_registry_client.rs [hunk 3]
```diff
@@ -0,0 +1,362 @@
...
+        || {
+            client.list_schema_versions(
+                GroupSelection::Default,
+                Some(schema_id.clone()),
+                None,
+                None,
+                TIMEOUT,
+            )
+        },
+        |versions| {
```
Comment (vaavva) [inline rust/azure_iot_operations_services/examples/edge_registry_client.rs]: For a sample, I would probably just have it print out the items in the list rather than treating this like a test and validating that the correct ones are there

File: rust/azure_iot_operations_services/examples/edge_registry_client.rs [hunk 4]
```diff
@@ -0,0 +1,362 @@
...
+        "labeled Thing Model was not listed",
+        LIST_RETRY_TIMEOUT,
+    )
+    .await?;
+
+    log::info!("Listed Thing Description {thing_description_id} and Thing Model {thing_model_id}");
+    Ok(())
+}
+
+#[cfg(test)]
```
Comment (vaavva) [inline rust/azure_iot_operations_services/examples/edge_registry_client.rs]: similar here, examples in the SDK are for the customers, they're not intended to be our e2e tests (would be great to add some of this to an e2e network test, but I don't think we have those set up yet for edge registry)

**Review by vaavva — APPROVED** [review]
Reviewed Rust Portions, skimmed .net portions. Rust SDK code looks great, have some comments for adjusting the sample

File: rust/azure_iot_operations_services/examples/edge_registry_client.rs (hunk 4 above)
Comment (bilalsellak) [reply rust/azure_iot_operations_services/examples/edge_registry_client.rs]: Ahh that makes sense. Ill update it

**Review by timtay-microsoft — APPROVED** [review]
.NET stuff LGTM

**Review by vaavva — APPROVED** [review]
